texthero/_helper.py

```diff
@@ -70,3 +71,242 @@ def wrapper(*args, **kwargs):
+    ```
+    then you know that the function takes a Pandas Series
+    whose cells are lists of strings (tokens) and will
+    return a Pandas Series whose cells are vectors of floats.
+    """
+
+    @staticmethod
+    def check_series():
+        raise NotImplementedError()  # Every Hero Series type has to have this.
+
+
+class TextSeries(HeroSeries):
+    """
+    In a TextSeries, every cell has to be a text, i.e. a string. For example,
+    `pd.Series(["test", "test"])` is a valid TextSeries.
+    """
+
+    @staticmethod
+    def check_series(s: pd.Series) -> bool:
+        """
+        Check if a given Pandas Series has the properties of a TextSeries.
+        """
+
+        error_string = (
+            "The input Series should consist only of strings in every cell."
+            " See help(hero.HeroSeries) for more information."
+        )
+
+        if not isinstance(s.iloc[0], str) or s.index.nlevels != 1:
+            raise TypeError(error_string)
+
+
+class TokenSeries(HeroSeries):
+    """
+    In a TokenSeries, every cell has to be a list of words/tokens, i.e. a list
+    of strings. For example, `pd.Series([["test"], ["token2", "token3"]])` is a valid TokenSeries.
+    """
+
+    @staticmethod
+    def check_series(s: pd.Series) -> bool:
+        """
+        Check if a given Pandas Series has the properties of a TokenSeries.
+        """
+
+        error_string = (
+            "There are non-token cells (every cell should be a list of words/tokens) in the given Series."
+            " See help(hero.HeroSeries) for more information."
+        )
+
+        def is_list_of_strings(cell):
+            return all(isinstance(x, str) for x in cell) and isinstance(
+                cell, (list, tuple)
+            )
+
+        if not is_list_of_strings(s.iloc[0]) or s.index.nlevels != 1:
+            raise TypeError(error_string)
+
+
+class VectorSeries(HeroSeries):
+    """
+    In a VectorSeries, every cell is a vector representing text, i.e.
+    a list of numbers.
+    For example, `pd.Series([[1.0, 2.0], [3.0]])` is a valid VectorSeries.
+    """
+
+    @staticmethod
+    def check_series(s: pd.Series, input_output="") -> bool:
+        """
+        Check if a given Pandas Series has the properties of a RepresentationSeries.
+        """
+
+        error_string = (
+            "There are non-representation cells (every cell should be a list of floats) in the given Series."
+            " See help(hero.HeroSeries) for more information."
+        )
+
+        def is_numeric(x):
+            try:
+                float(x)
+            except ValueError:
+                return False
+            else:
+                return True
+
+        def is_list_of_numbers(cell):
+            return all(is_numeric(x) for x in cell) and isinstance(cell, (list, tuple))
+
+        if not is_list_of_numbers(s.iloc[0]) or s.index.nlevels != 1:
+            raise TypeError(error_string)
+
+
+class DocumentRepresentationSeries(HeroSeries):
```

Comment:
I see why you called it `Document..` but would rather keep it more general: `RepresentationSeries`

Example: word embeddings. This is can also be seen as `RepresentationSeries`, the first index is the word and the second index is the `dim_x` (but it's also true that this can also be a `VectorSeries` ). Having a separate Series for this would not be necessary...

If we end-up seeing we want to distinguish between the twos, then we might create other two classes `DocRepresentationSeries` and `WordRepresentationSeries` and inherit `RepresentationSeries` ... but then we might also need `TokenRepresentationSeries` ... -> `RepresentationSeries` should be enough.

Reply:
Makes sense, will do